parser-typechecker/src/Unison/CommandLine/OutputMessages.hs

```diff
@@ -323,7 +324,25 @@ notifyUser dir o = case o of
     else putPretty' "  🚫  "
     pure mempty
 
-  LinkFailure input -> pure $ P.warnCallout . P.shown $ input
+  MetadataMissingType ppe ref -> pure . P.fatalCallout . P.lines $ [
+    P.wrap $ "The metadata value " <> P.red (prettyTermName ppe ref)
+          <> "is missing a type signature in the codebase.",
+    "",
+    P.wrap $ "This might be due to pulling an incomplete"
+          <> "or invalid codebase, or because files inside the codebase"
+          <> "are being deleted external to UCM."
+    ]         
+  MetadataAmbiguous ppe [] -> pure . P.warnCallout .
+    P.wrap $ "Nothing to do. I couldn't find any matching metadata."
+  MetadataAmbiguous ppe refs -> pure . P.warnCallout . P.lines $ [
+    P.wrap $ "I'm not sure which metadata value you're referring to"
+          <> "since there are multiple matches:",
+    "",
+    P.indentN 2 $ P.spaced (P.blue . prettyTermName ppe <$> refs),
+    "",
+    tip "Try again and supply one of the above definitions explicitly."
+    ]
```

Comment:
Thought about doing numbering for these, but didn't bother with that yet and wasn't sure it was a good idea. The `docs` and `links` commands already give you a numbered list.